### PR TITLE
chore: fix eslint and ts errors in dev and introduce tsconfig.prod.ts file

### PR DESCRIPTION
**Pull request by mkarajohn**
Resolves [DSTRBTN-540]

<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## TL;DR
> This pull request updates the ESLint configuration, package.json, and TypeScript configuration files to improve the build process and linting rules.
> 
> ## What changed
> - Updated ESLint rules to disable unnecessary rules for React 17+ and added TypeScript specific rules under the overrides section.
> - Added "types" to the exports field in package.json.
> - Updated the build scripts in package.json to use a new tsconfig.prod.json file.
> - Simplified the tsconfig.json file and created a new tsconfig.prod.json file for production builds.
> - The tsconfig.prod.json file excludes test files from the build process.
> 
> ## How to test
> 1. Pull down the changes from this branch.
> 2. Run `npm install` to update your packages.
> 3. Run `npm run lint` to check the linting rules.
> 4. Run `npm run build` to test the build process.
> 
> ## Why make this change
> These changes improve the build process by excluding test files and improve the linting rules to better match our usage of React and TypeScript. This will help maintain code quality and reduce build times.
</details>

[DSTRBTN-540]: https://orfium.atlassian.net/browse/DSTRBTN-540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mkarajohn** [comment]
Current dependencies on/for this PR:
* master
  * **PR #74** <a href="https://app.graphite.dev/github/pr/Orfium/toolbox/74" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Orfium/toolbox/74?utm_source=stack-comment).

File: tsconfig.json [hunk 1]
```diff
@@ -3,21 +3,15 @@
...
+    "lib": ["es5", "es6", "es7", "es2017", "dom"],
     "sourceMap": true,
     "skipLibCheck": true,
     "allowJs": true,
     "jsx": "react-jsx",
     "esModuleInterop": true,
     "moduleResolution": "node",
     "strict": true,
-    "rootDir": "./src",
+    "rootDir": "./",
```
Comment (panvourtsis) [inline tsconfig.json:14]: why we changed this to root?

**Review by panvourtsis — DISMISSED** [review]

File: tsconfig.json (hunk 1 above)
Comment (mkarajohn) [reply tsconfig.json:14]: This was changed in order to stop ts errors when we import from outside of the `src/` directory in our tests, if I recall correctly. The production config sets it back to `src/`

Comment (mkarajohn) [reply tsconfig.json:14]: Resolving this

**Pull request closed without merging**